tests/sycl/interop_handle.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of hipSYCL, a SYCL implementation based on CUDA/HIP
+ *
+ * Copyright (c) 2018-2023 Aksel Alpay and contributors
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#include "sycl_test_suite.hpp"
+#include <boost/test/unit_test_suite.hpp>
+
+BOOST_FIXTURE_TEST_SUITE(interop_handle_tests, reset_device_fixture)
+
+BOOST_AUTO_TEST_CASE(interop_handle_api) {
+  namespace s = cl::sycl;
+  using namespace hipsycl;
+
+  sycl::device d;
+
+  try {
+    d = sycl::device(sycl::gpu_selector_v);
+  }
+  catch (sycl::exception const &e) {
+    d = sycl::device(sycl::cpu_selector_v);
+  }
+
+  s::queue q(d);
+  
+  rt::device_id assigned_device{rt::backend_descriptor{rt::hardware_platform::cpu,
+                                rt::api_platform::omp}, 12345};
+
+  rt::backend_executor *executor = nullptr;
+  s::interop_handle ih{assigned_device, executor};
+  s::backend b = ih.get_backend();
+
+  BOOST_CHECK(b == q.get_device().get_backend());
```

Comment:
This is not correct. `b` should always be the OpenMP backend because that's what you use for `assigned_device`. `q` may however be constructed using an arbitrary different backend.

Suggestion: You don't need `q`, just remove it and check whether `b` == s::backend::omp`.